test/web-platform-tests/run-tuwpts.js

```diff
@@ -21,18 +21,20 @@ const possibleTestFilePaths = getPossibleTestFilePaths(manifest);
 let wptServerURL;
 let serverProcess;
 const runSingleWPT = require("./run-single-wpt.js")(() => wptServerURL);
-before({ timeout: 30 * 1000 }, async () => {
+const wptServerPromise = (async () => {
   const { urls, subprocess } = await startWPTServer({ toUpstream: true });
   wptServerURL = urls[0];
   serverProcess = subprocess;
-});
-
-after(() => {
-  serverProcess.kill();
-});
+})();
 
 describe("Local tests in web-platform-test format (to-upstream)", () => {
+  before({ timeout: 30 * 1000 }, () => wptServerPromise);
+
   for (const test of possibleTestFilePaths) {
     runSingleWPT(test);
   }
+
+  after(() => {
```

Comment:
The `after` should still be outside the `describe`, as we are starting the WPT server unconditionally regardless if `to-upstream` tests are actually run. This bug might be reproducible if you try `mocha --grep=something-thats-not-wpt` or something similar.

When changing that, you might want to further make sure `serverProcess` is not undefined before you try to call `kill()` on it.

Reply:
I decided to make the WPT subprocess available synchronously in the same way as `util.promisify(childProcess.exec)` behaves.

Reply:
Works for me.